PancakeSwap-Bots/Auto-CAKE-Admin-Operations/src/agent.spec.ts

```diff
@@ -0,0 +1,168 @@
+import { HandleTransaction, ethers, Finding } from "forta-agent";
+import { TestTransactionEvent, createAddress, TraceProps } from "forta-agent-tools/lib/tests";
+import { NetworkManager } from "forta-agent-tools";
+
+import bot from "./agent";
+
+import { EVENTS, ABI } from "./abi";
+import { createEventFinding, createFunctionFinding } from "./findings";
+
+describe("CakeVault", () => {
+  let handleTransaction: HandleTransaction;
+
+  let functionInterface: ethers.utils.Interface;
+
+  let mockTxEvent: TestTransactionEvent;
+
+  const MOCK_CONTRACT_ADDRESS = createAddress("0x0123");
+
+  let mockEventFragment: ethers.utils.EventFragment;
+  let eventFragments: ethers.utils.EventFragment[] = [];
+  let functionFragments: ethers.utils.FunctionFragment[] = [];
+
+  interface NetworkData {
+    cakeVaultAddress: string;
+  }
+
+  let networkManager: NetworkManager<NetworkData>;
+
+  beforeEach(() => {
+    mockTxEvent = new TestTransactionEvent();
+  });
+
+  beforeAll(() => {
+    const mockData: Record<number, NetworkData> = {
+      111: {
+        cakeVaultAddress: MOCK_CONTRACT_ADDRESS,
+      },
+    };
+
+    networkManager = new NetworkManager(mockData, 111);
+    handleTransaction = bot.provideHandleTransaction(networkManager);
+
+    functionInterface = new ethers.utils.Interface(ABI);
+
+    EVENTS.forEach((event) => {
+      eventFragments.push(ethers.utils.EventFragment.fromString(event.slice("event ".length)));
+    });
+
+    ABI.forEach((func) => {
+      functionFragments.push(ethers.utils.FunctionFragment.fromString(func.slice("function ".length)));
+    });
+
+    mockEventFragment = ethers.utils.EventFragment.from("MockEvent(uint256)");
+  });
+
+  describe("handleTransaction", () => {
```

Comment:
This describe is unnecessary, the unit tests can be at the same scope as the `beforeEach()` and `beforeAll()`.

Reply:
The code block inside of this `describe("handleTransaction",...)` is not needed since you already have a parent `describe("CakeVault Test Suite",...)`.

Seldom do we have a `describe()` inside of another `describe()`. Therefore, you can move the code block out of `describe("handleTransaction",...)` to be at the same scope as `beforeEach()` and `beforeAll()` and delete `describe("handleTransaction,..)`.